compiler/rustc_smir/src/rustc_internal/mod.rs

```diff
@@ -54,139 +43,11 @@ pub fn stable<'tcx, S: Stable<'tcx>>(item: S) -> S::T {
 /// # Panics
 ///
 /// This function will panic if StableMIR has not been properly initialized.
-pub fn internal<'tcx, S>(tcx: TyCtxt<'tcx>, item: S) -> S::T<'tcx>
+pub fn internal<'tcx, S>(item: S) -> S::T<'tcx>
```

Comment:
Actually, this isn't safe. This allows users to create objects with any lifetime `'tcx`, including `'static`. See [this commit](https://github.com/rust-lang/rust/pull/120128/commits/9aace6723572438a94378451793ca37deb768e72) and [this PR](https://github.com/rust-lang/rust/pull/120128) for more context.